client/landing/stepper/declarative-flow/internals/steps-repository/launchpad/sidebar.tsx

```diff
@@ -77,6 +77,14 @@ const Sidebar = ( { sidebarDomain, siteSlug, submit, goNext, goToStep, flow }: S
 
 	const currentTask = getTasksProgress( enhancedTasks );
 	const launchTask = enhancedTasks?.find( ( task ) => task.isLaunchTask === true );
+
+	// Free flow - remove domain_upsell task if user is on paid plan
+	if ( flow === FREE_FLOW && ! site?.plan?.is_free && enhancedTasks ) {
```

Comment:
Similar comment below, but we should make this upsell appear for all launchpad flows where the upsell is relevant. We can start with free, write, and build to keep testing simple and get this out sooner. But follow up to add and test the other flows as well.